embrace-gradle-plugin/src/main/java/io/embrace/android/gradle/plugin/tasks/ndk/HashSharedObjectFilesTask.kt

```diff
@@ -0,0 +1,64 @@
+package io.embrace.android.gradle.plugin.tasks.ndk
+
+import io.embrace.android.gradle.plugin.hash.calculateSha1ForFile
+import io.embrace.android.gradle.plugin.tasks.EmbraceTaskImpl
+import io.embrace.android.gradle.plugin.util.serialization.MoshiSerializer
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * Given a directory containing compressed shared object files organized by architecture,
+ * this task calculates SHA1 hashes for each compressed file and creates a JSON mapping
+ * of architectures to their corresponding hashes.
+ */
+abstract class HashSharedObjectFilesTask @Inject constructor(
+    objectFactory: ObjectFactory,
+) : EmbraceTaskImpl(objectFactory) {
+
+    private val serializer = MoshiSerializer()
+
+    @get:InputDirectory
+    @get:SkipWhenEmpty
+    val compressedSharedObjectFilesDirectory: DirectoryProperty = objectFactory.directoryProperty()
+
+    @get:OutputFile
+    val architecturesToHashedSharedObjectFilesMap: RegularFileProperty = objectFactory.fileProperty()
+
+    @TaskAction
+    fun onRun() {
+        val outputMap = createOutputMap()
+
+        // Serialize the map to JSON and write it to the output file
+        val serializableMap = ArchitecturesToHashedSharedObjectFilesMap(outputMap)
+        architecturesToHashedSharedObjectFilesMap.get().asFile.outputStream().use { outputStream ->
+            serializer.toJson(serializableMap, ArchitecturesToHashedSharedObjectFilesMap::class.java, outputStream)
+        }
+    }
+
+    private fun createOutputMap(): Map<String, Map<String, String>> =
+        compressedSharedObjectFilesDirectory.get().asFile
+            .listFiles()
+            ?.filter { it.isDirectory && it.listFiles()?.isNotEmpty() == true }
```

Comment:
nit: consider something like `?: emptyList()` to avoid the null coalescing operator in this chain